### ISSUE TITLE
New component: cgroup aware go runtime extension

### ISSUE THREAD

**Issue by tomershafir** [open-telemetry/opentelemetry-collector-contrib#30289]
### The purpose and use-cases of the new component

Set go runtime variables based on linux cgroupfs automatically, or let the user set a cgroup relative value. For example, set GOMAXPROCS and GOMEMLIMIT by importing https://github.com/uber-go/automaxprocs and https://github.com/KimMachineGun/automemlimit.

### Example configuration for the component

Proxy config for https://github.com/uber-go/automaxprocs and https://github.com/KimMachineGun/automemlimit

### Telemetry data types supported

It is data type independent

### Is this a vendor-specific component?

- [ ] This is a vendor-specific component
- [ ] If this is a vendor-specific component, I am proposing to contribute and support it as a representative of the vendor.

### Sponsor (optional)

@mx-psi

### Additional context

I think about a single extension that should theoretically cover the entire go runtime, starting with the 2 variables mentioned above.

Core issue ref: https://github.com/open-telemetry/opentelemetry-collector/issues/9203

**Comment by mx-psi** [comment]
cc @open-telemetry/helm-maintainers would love to know what you think (is this something we would want on the Helm chart? If so, anything specific about the design we should take into account?)

**Comment by TylerHelmuth** [comment]
Seems reasonable to me.  In the helm chart an extension like this would allow us to remove some templates like https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/_helpers.tpl#L169.  A solution like this also helps users running the collector outside of kubernetes.

I'd like to see some example configs of how the extension would be configured.

**Comment by tomershafir** [comment]
maybe something like:
```
extensions:
  cgroupgoruntime:
    gomaxprocs: # hardcoded percentage 100
      enable: true
    gomemlimit:
      percentage: 85 # default 90, enabled by value > 0
```

**Comment by TylerHelmuth** [comment]
Would `gomaxprocs` or `gomemlimit` ever have additional fields in their sections?

**Comment by tomershafir** [comment]
I assume that the linux and go relevant interfaces should be stable for long, so I guess no. There may be additional sections.

**Comment by cforce** [comment]
Setting gomaxprocs as env var does not reduce occurs as expected .

**Comment by iblancasa** [comment]
It would be interesting to have this extension. Recently, [Prometheus](https://github.com/prometheus/prometheus/releases/tag/v2.50.0) added automatic memory limit handling based on the same mechanism.

**Comment by mx-psi** [comment]
@tomershafir no, but I can remove the `Stale` marker and leave it open for now in case somebody else volunteers

**Comment by mx-psi** [comment]
Just to clarify: I am still willing to sponsor this component :)

**Comment by rogercoll** [comment]
I think this issue can be closed as the component was added in the code base https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35472

Also, about to be added in the contrib release https://github.com/open-telemetry/opentelemetry-collector-releases/pull/789
